### ISSUE TITLE
No error for nonlocal with a missing variable

### ISSUE THREAD

**Issue by eltoder** [astral-sh/ruff#21495]
### Summary

The `nonlocal` statement requires that the variable is a local variable in one of the parent scopes. ruff does not check this. For example:
```python
def no_local_var():
    def setter(xx):
        nonlocal x
        x = xx
        print(x)

    return setter
```
(playground link: https://play.ruff.rs/f6ca3be7-208f-4ade-b68e-148d3546af6b)

This passes all checks:
```
$ ruff check --isolated t2.py
All checks passed!
```
However this fails with a SyntaxError when run:
```
$ python t2.py
  File "/home/eltoder/dev/scratch/t2.py", line 3
    nonlocal x
    ^^^^^^^^^^
SyntaxError: no binding for nonlocal 'x' found
```

### Version

0.14.5

**Comment by MichaReiser** [comment]
Hi

We plan on [making this a normal syntax error](https://github.com/astral-sh/ruff/issues/17412), but, for now, you have to enable [PLE0117](https://docs.astral.sh/ruff/rules/nonlocal-without-binding/) to catch this ([playground](https://play.ruff.rs/49728d2e-065c-41fc-b254-f4788017f77f))

**Comment by eltoder** [comment]
I see. I think it would be better if you enabled these errors as soon as they are implemented, but I'm sure you have your reasons.
